## Load libraries

```{r}
#| message: false
library(readr)
library(tidyverse)
```

## Set working directory

```{r}
project_path <- "~/projects/group_11_project/"
```

## Load data

```{r}
<<<<<<< HEAD
expr_data <- read_csv(str_c(project_path, "data/_raw/Colorectal_Cancer_Gene_Expression_Data.csv"))
patient_data <- read_csv(str_c(project_path, "data/_raw/Colorectal_Cancer_Patient_Data.csv"))

=======
expr_data <- read_csv("~/Projects/group_11_project/data/_raw/Colorectal_Cancer_Gene_Expression_Data.csv")
patient_data <- read_csv("~/Projects/group_11_project/data/_raw/Colorectal_Cancer_Patient_Data.csv")
>>>>>>> d0dfb48756405c1eea7edaa33d712e9f960f8fda
```

```{r}
# FPKM normalized gene expression data:
# expr_data <- read_tsv("https://www.ncbi.nlm.nih.gov/geo/download/?type=rnaseq_counts&acc=GSE50760&format=file&file=GSE50760_norm_counts_FPKM_GRCh38.p13_NCBI.tsv.gz")

# Human gene annotation data:
# annotation <- read_tsv("https://www.ncbi.nlm.nih.gov/geo/download/?format=file&type=rnaseq_counts&file=Human.GRCh38.p13.annot.tsv.gz")
```

## Combine the two datasets

```{r}
# Pivot expr_data
expr_data <- expr_data |>
  mutate(ID_GENE = ID_REF) |> 
  select(-ID_REF) |> 
  pivot_longer(-ID_GENE, names_to = "ID_REF", values_to = "Gene expression level") |> 
  pivot_wider(names_from = ID_GENE, values_from = "Gene expression level") 

# Join the two datasets
data <- patient_data |>
  full_join(expr_data, by = "ID_REF")

write_tsv(data, 
          str_c(project_path, "data/01_dat_load.tsv"))
```
